pennylane/grouping/utils.py

```diff
@@ -658,6 +658,38 @@ def is_qwc(pauli_vec_1, pauli_vec_2):
     return True
 
 
+def are_pauli_words_qwc(lst_pauli_words):
+    """Given a list of observables assumed to be valid Pauli words, determine if they
+     are all mutually qubit-wise commuting.
+
+    This implementation has time complexity ~ O(m * n) for n wires and m Pauli words.
+
+    Args:
+        lst_pauliwords (list[Observable]): list of observables (assumed to be valid Pauli words).
+
+    Returns:
+        (bool): True if they are all qubit-wise commuting, false otherwise.
+    """
+    wire_dict = {}
```

Comment:
Worth making this name more descriptive or adding a comment on what the object is used for.

E.g., later when reading `wire_dict[wire]`, the reader should be able to tell what the value of that expression is. Right now it's not apparent.
```suggestion
    latest_op_name_per_wire = {}
```

Reply:
Added!

Reply:
Great catch on the code readability there, I would have totally missed it. But looking back this is much nicer